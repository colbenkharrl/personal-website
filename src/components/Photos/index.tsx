import Image from 'next/image';
import clsx from 'clsx';

import image2 from '@/images/photos/image-2.jpg';
import image3 from '@/images/photos/image-3.jpg';
import image4 from '@/images/photos/image-4.jpg';

export function Photos() {
  return (
    <div className="mt-16 sm:mt-20" data-testid="Photos">
      <div className="-my-4 flex justify-center gap-5 overflow-hidden py-4 sm:gap-8">
        {[image2, image3, image4].map((image, imageIndex) => (
          <div
            key={image.src}
            className={clsx(
              'relative aspect-[1/1] w-28 flex-none overflow-hidden rounded-xl bg-zinc-100 dark:bg-zinc-800 sm:w-40 sm:rounded-2xl md:w-48 lg:w-56 xl:w-64 2xl:w-72',
            )}
          >
            <Image
              src={image}
              alt=""
              sizes="(min-width: 640px) 18rem, 11rem"
              className="absolute inset-0 h-full w-full object-cover"
            />
          </div>
        ))}
      </div>
    </div>
  );
}
