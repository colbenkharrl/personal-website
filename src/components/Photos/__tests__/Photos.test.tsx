import { render, screen } from '@testing-library/react';
import * as stories from '../__stories__/Photos.stories';
import { composeStories } from '@storybook/react';

const { Basic } = composeStories(stories);

it('Photos matches snapshot (regression test)', () => {
  render(<Basic data-testid="Photos" />);
  expect(screen.getByTestId('Photos')).toMatchSnapshot(
    'Full component snapshot.',
  );
});
