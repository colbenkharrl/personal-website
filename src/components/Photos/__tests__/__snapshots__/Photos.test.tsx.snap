// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photos matches snapshot (regression test): Full component snapshot. 1`] = `
<div
  class="mt-16 sm:mt-20"
  data-testid="Photos"
>
  <div
    class="-my-4 flex justify-center gap-5 overflow-hidden py-4 sm:gap-8"
  >
    <div
      class="relative aspect-[1/1] w-28 flex-none overflow-hidden rounded-xl bg-zinc-100 dark:bg-zinc-800 sm:w-40 sm:rounded-2xl md:w-48 lg:w-56 xl:w-64 2xl:w-72"
    >
      <img
        alt=""
        class="absolute inset-0 h-full w-full object-cover"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        sizes="(min-width: 640px) 18rem, 11rem"
        src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=16&q=75 16w, /_next/image?url=%2Fimg.jpg&w=32&q=75 32w, /_next/image?url=%2Fimg.jpg&w=48&q=75 48w, /_next/image?url=%2Fimg.jpg&w=64&q=75 64w, /_next/image?url=%2Fimg.jpg&w=96&q=75 96w, /_next/image?url=%2Fimg.jpg&w=128&q=75 128w, /_next/image?url=%2Fimg.jpg&w=256&q=75 256w, /_next/image?url=%2Fimg.jpg&w=384&q=75 384w, /_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
        style="color: transparent;"
        width="40"
      />
    </div>
    <div
      class="relative aspect-[1/1] w-28 flex-none overflow-hidden rounded-xl bg-zinc-100 dark:bg-zinc-800 sm:w-40 sm:rounded-2xl md:w-48 lg:w-56 xl:w-64 2xl:w-72"
    >
      <img
        alt=""
        class="absolute inset-0 h-full w-full object-cover"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        sizes="(min-width: 640px) 18rem, 11rem"
        src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=16&q=75 16w, /_next/image?url=%2Fimg.jpg&w=32&q=75 32w, /_next/image?url=%2Fimg.jpg&w=48&q=75 48w, /_next/image?url=%2Fimg.jpg&w=64&q=75 64w, /_next/image?url=%2Fimg.jpg&w=96&q=75 96w, /_next/image?url=%2Fimg.jpg&w=128&q=75 128w, /_next/image?url=%2Fimg.jpg&w=256&q=75 256w, /_next/image?url=%2Fimg.jpg&w=384&q=75 384w, /_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
        style="color: transparent;"
        width="40"
      />
    </div>
    <div
      class="relative aspect-[1/1] w-28 flex-none overflow-hidden rounded-xl bg-zinc-100 dark:bg-zinc-800 sm:w-40 sm:rounded-2xl md:w-48 lg:w-56 xl:w-64 2xl:w-72"
    >
      <img
        alt=""
        class="absolute inset-0 h-full w-full object-cover"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        sizes="(min-width: 640px) 18rem, 11rem"
        src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=16&q=75 16w, /_next/image?url=%2Fimg.jpg&w=32&q=75 32w, /_next/image?url=%2Fimg.jpg&w=48&q=75 48w, /_next/image?url=%2Fimg.jpg&w=64&q=75 64w, /_next/image?url=%2Fimg.jpg&w=96&q=75 96w, /_next/image?url=%2Fimg.jpg&w=128&q=75 128w, /_next/image?url=%2Fimg.jpg&w=256&q=75 256w, /_next/image?url=%2Fimg.jpg&w=384&q=75 384w, /_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
        style="color: transparent;"
        width="40"
      />
    </div>
  </div>
</div>
`;
